### PR TITLE
Update component version to 10.4.1, GeoEvent dependency to 10.4.1+, ActiveMQ to 5.16.0

### PR DESCRIPTION
**Pull request by brudo**
This is mostly a dependency update:

- OSGI bundle version updated to 10.4.1 (also the component version in GeoEvent Manager)

- ActiveMQ version 5.16.0 (removing commented-out dependencies required for embedded peer or vm broker)

(comment updated to reflect current state)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,8 +16,11 @@
   <properties>
     <contact.address>geoevent@esri.com</contact.address>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.bundle.plugin.version>3.5.0</maven.bundle.plugin.version>
-    <activemq.version>5.15.10</activemq.version>
+    <maven.bundle.plugin.version>4.2.1</maven.bundle.plugin.version>
+
+    <activemq.version>5.16.0</activemq.version>
+
+    <geoevent.compat.version>10.4.1</geoevent.compat.version>
```
Comment (hanoch) [inline pom.xml]: we don't need that

File: pom.xml [hunk 2]
```diff
@@ -35,15 +40,38 @@
     <dependency>
       <groupId>com.esri.geoevent.sdk</groupId>
       <artifactId>geoevent-sdk</artifactId>
-      <version>${project.version}</version>
+      <version>[{geoevent.compat.version},)</version>
```
Comment (hanoch) [inline pom.xml]: please revert this. @eironside please review.

File: pom.xml [hunk 3]
```diff
@@ -35,15 +40,38 @@
...
+      <scope>provided</scope>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.activemq</groupId>
+      <artifactId>activemq-jdbc-store</artifactId>
+      <version>${activemq.version}</version>
+      <scope>provided</scope>
+      <optional>true</optional>
+    </dependency>
```
Comment (hanoch) [inline pom.xml]: why are these needed?

File: pom.xml [hunk 4]
```diff
@@ -35,15 +40,38 @@
...
+      <version>[{geoevent.compat.version},)</version>
       <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.activemq</groupId>
       <artifactId>activemq-client</artifactId>
       <version>${activemq.version}</version>
     </dependency>
-    <!-- dependencies for embedding via the vm and peer transports omitted -->
+    <!-- dependencies for embedding via the vm and peer transports included -->
```
Comment (hanoch) [inline pom.xml]: ?

File: pom.xml [hunk 5]
```diff
@@ -14,8 +16,11 @@
   <properties>
     <contact.address>geoevent@esri.com</contact.address>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.bundle.plugin.version>3.5.0</maven.bundle.plugin.version>
-    <activemq.version>5.15.10</activemq.version>
+    <maven.bundle.plugin.version>4.2.1</maven.bundle.plugin.version>
+
+    <activemq.version>5.16.0</activemq.version>
```
Comment (hanoch) [inline pom.xml:18]: that one might be okay (assumimng it was tested)

**Review by hanoch — CHANGES_REQUESTED** [review]
it is okay to update the AMQ lib version to

File: pom.xml (hunk 1 above)
Comment (brudo) [reply pom.xml]: This preserves backward compatibility and has no down side that I know of. Why wouldn't we?

Comment (hanoch) [reply pom.xml]: We should build this connector as a single version throughout the pom, java and xml files - 10.8.1, 10.4.1, 10.4, 10.6, or other.
@eironside is managing all of these connector versions now and has a standard in place that we need all of them to follow.

File: pom.xml (hunk 2 above)
Comment (brudo) [reply pom.xml]: As noted above, this preserves backward compatibility and has no down side that I know of. I look forward to comment from @eironside as well.

If you reject these two changes, then I will need to set the package version to 10.7.1, as it is critical that it works at least back to that version. In that case, unless there is a reason not to do this, I'd still prefer to use the [v,) notation to specify the minimum required SDK version as 10.7.1, and indicate that later versions are also acceptable (even though it will probably work with later versions of GeoEvent Server even without that).

File: pom.xml (hunk 4 above)
Comment (brudo) [reply pom.xml]: So, in addition to functioning as a client to an ActiveMQ broker listening elsewhere, the ActiveMQ client also includes the ability run a broker within the calling process, just by using special URIs. I have tested this and it does work. However, if the required dependencies are not available, it fails.

For more info see here https://activemq.apache.org/activemq-connection-uris under the headings for VM transport, Peer transport, and Broker Configuration URIs.

If you don't want this functionality in the published version of ActiveMQ for GeoEvent, I'm OK with that - I am not using that functionality in production, and I can keep those extra dependency references in a branch that I maintain separately instead, in case needed for tests or demos.

File: pom.xml (hunk 3 above)
Comment (brudo) [reply pom.xml]: When configuring an embedded broker (see next review comment below), it is possible to configure that broker as non-persistent by specify persistent=false in the URI. However, if you don't specify that, and also don't configure the persistence mechanism, failures will occur.

By including these additional dependencies, we can configure an embedded broker that persists to either a file based journal or a database. This works (tested years ago), but it is a bit tricky to get right.

As noted in the comments in the POM, I do consider the support for embedded brokers to be experimental, even though it is based on supported and documented functionality of the ActiveMQ client. In most cases I prefer to run a broker that is not embedded, especially if persistence is needed.

File: pom.xml (hunk 1 above)
Comment (brudo) [reply pom.xml]: OK, happy to support whatever standard @eironside is using to maintain these, once I am clear on what is needed.

**Comment by brudo** [comment]
OK, I will pull out the three embedded broker reference and move a separate branch. Then will delete them from this branch.

File: pom.xml (hunk 2 above)
Comment (brudo) [reply pom.xml]: I am keeping this part unchanged in my branch for now until I understand better from @eironside about the standard.

Comment (brudo) [reply pom.xml]: Actually I changed the project version to 10.4.1 and removed geoevent.compat.version. The explicit [v,) around the version number is recommended in the maven docs, even though the implicit behavior for v does seem to behave the same in GeoEvent Server. I tested this in GeoEvent Server 10.7.1 and confirmed it works. I have not yet tested with 10.8.1.

File: pom.xml (hunk 4 above)
Comment (brudo) [reply pom.xml]: note this section is deleted with the second commit; it is not required for normal usage and reduces the JAR size by 3MB. i placed the embedded broker dependencies in a separate branch if would like to talk about it separately, but for this PR that should resolve the ?

File: pom.xml (hunk 1 above)
Comment (brudo) [reply pom.xml]: In the end, having switched the project version to 10.4.1 and removed the compat version in the second commit, and tested with 10.7.1, I agree we don't need that.

File: pom.xml [hunk 6]
```diff
@@ -35,16 +37,14 @@
     <dependency>
       <groupId>com.esri.geoevent.sdk</groupId>
       <artifactId>geoevent-sdk</artifactId>
-      <version>${project.version}</version>
+      <version>[{project.version},)</version>
```
Comment (hanoch) [inline pom.xml]: what is the change here?

Comment (brudo) [reply pom.xml]: Thank you for pointing it out; there is a typo there - $ omitted - although it still worked, I will fix that. The ${v} => [${v},) just explicitly indicates that future versions are supported, as recommended by Maven. Considering that 10.4.1 is quite an old release, someone perusing the source code for indications of forward compatibility might find that reassuring, even though it does match the default behavior for most cases. I don't insist on it; I just thought it was a good idea.

Comment (hanoch) [reply pom.xml]: can we keep it simply as  ```<version>${project.version}</version>```?
We don't know which future version might not be supported.

**Review by hanoch — APPROVED** [review]

**Comment by brudo** [comment]
@hanoch we requested @eironside to review - I think that is all that needs to be closed off before it could be merged. With the revisions, I imagine it should be fine now though.

File: pom.xml [hunk 7]
```diff
@@ -5,7 +5,9 @@
 
   <groupId>com.esri.geoevent.parent</groupId>
   <artifactId>activemq</artifactId>
-  <version>10.6.1</version>
+
```
Comment (hanoch) [inline pom.xml]: can we remove this extra line here?

Comment (brudo) [reply pom.xml]: Done

File: pom.xml (hunk 5 above)
Comment (brudo) [reply pom.xml:18]: It was tested and is working with 10.7.1. Still have not been able to test with 10.8.1.
